shell/platform/darwin/macos/framework/Source/FlutterSurface.mm

```diff
@@ -0,0 +1,94 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterSurface_Internal.h"
+
+#import <Metal/Metal.h>
+
+@interface FlutterSurface () {
+  CGSize _size;
+  IOSurfaceRef _ioSurface;
+  id<MTLTexture> _texture;
+}
+@end
+
+@implementation FlutterSurface
+
+- (IOSurfaceRef)ioSurface {
+  return _ioSurface;
+}
+
+- (CGSize)size {
+  return _size;
+}
+
+- (int64_t)textureId {
+  return reinterpret_cast<int64_t>(_texture);
+}
+
+- (instancetype)initWithSize:(CGSize)size device:(id<MTLDevice>)device {
+  if (self = [super init]) {
+    self->_size = size;
+    self->_ioSurface = [FlutterSurface createIOSurfaceWithSize:size];
+    self->_texture = [FlutterSurface createTextureForIOSurface:_ioSurface size:size device:device];
+  }
+  return self;
+}
+
+static void ReleaseSurface(void* surface) {
```

Comment:
It gets little bit ugly because of ARC. I think it'd need to be something like this. Every cast needs a __bridge.

```objc
static void ReleaseSurface(void* user_data) {
  FlutterSurface* surface = (__bridge id)user_data;
  if (surface != nullptr) {
    CFBridgingRelease((__bridge void*)surface);
  }
}
```

or maybe 

```objc
static void ReleaseSurface(void* user_data) {
  if (user_data != nullptr) {
    FlutterSurface* surface = CFBridgingRelease(user_data);
    (void)surface;
  }
}
```

Which one do you prefer?

Reply:
Ugh yeah, the ARC implication hadn't crossed my mind. 

Of the two, the first reads a little better to me, mostly since people will be less likely to simplify it, the go through the whole thing of hitting the unused var warning, then removing the local that was only there to show the type, at which point we're left with the same thing with a different variable name.

To be honest, after all that, I'd be tempted to just leave as-is. I don't think the very minor hinting we'd get is worth the tradeoff.